src/Resources/contao/dca/tl_article.php

```diff
@@ -536,23 +534,13 @@ public function checkPermission()
 											  ->execute($id);
 
 					// Check whether the current user has permission for the current page
-					if (!$error && $objPage->numRows)
+					if ($objPage->numRows && !$this->User->isAllowed($permission, $objPage->row()))
 					{
-						if (!$this->User->isAllowed($permission, $objPage->row()))
-						{
-							$this->log('Not enough permissions to '. Input::get('act') .' '. (strlen(Input::get('id')) ? 'article ID '. Input::get('id') : ' articles') .' on page ID '. $id .' or paste it/them into page ID '. $id, __METHOD__, TL_ERROR);
-							$error = true;
-						}
+						throw new Contao\CoreBundle\Exception\AccessDeniedException('Not enough permissions to ' . Input::get('act') . ' ' . (strlen(Input::get('id')) ? 'article ID ' . Input::get('id') : ' articles') . ' on page ID ' . $id . ' or to paste it/them into page ID ' . $id . '.');
 					}
 				}
 			}
 		}
-
-		// Redirect if there is an error
-		if ($error)
```

Comment:
Btw. wasn't there a reason this was handled later instead of instantly redirecting?